### DISCUSSION TITLE
Manual Install of Older version

### DISCUSSION THREAD

**Discussion by pdieng** [stm32duino/Arduino_Core_STM32#2456] [Q&A] [answered]
I am trying to execute another person's project that uses the STM32duino core.  Since the code is from 2013, there are features in the code that are no longer in the core version 2.0.0-up and causes me to get compiling errors.

So I can either try and get the old version of the core that is compatible with the firmware or rewrite the necessary parts of the firmware to update it to a current version of the core.

My first attempt is to grab an older version to try and get the existing code to compile but I cant' seem to get the library to install.  I am grabbing the Source for v1.8.0 from here:
https://github.com/stm32duino/Arduino_Core_STM32/releases?page=2

I unzipped and pasted the folder it into my Sketchbook directory but when I open Arduino 1.8.13, it tells me that it is an invalid library because there are "no headers files".

How can I get a package of v1.8.0 to install in arduino or is it possible?

**Comment by fpistm** [comment]
Why unzip? Simply use the install from json file. Or it is another core?

**Reply by fpistm** [reply, marked as answer]
To get the 1.8.0 you can use this package index:
https://raw.githubusercontent.com/stm32duino/BoardManagerFiles/master/STM32/package_stm_index.json

**Reply by pdieng** [reply]
I didn't know how to get the json and didn't know it was even available.  How would I find that json link if you hadn't given it to me?

I added that json to the additional boards and I am now able to install older versions!!!

Now it is just a matter of figuring out how far back I need to go.

Thanks for your help!!

**Reply by fpistm** [reply]
Well it is the same repository so all history is kept. This is the goal of GitHub.
